tests/pytests/functional/states/test_npm.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+from salt.exceptions import CommandExecutionError
+
+
+@pytest.fixture(scope="module", autouse=True)
+def install_npm(sminion):
+    try:
+        sminion.functions.pkg.install("npm")
+        # Just name the thing we're looking for
+        sminion.functions.npm  # pylint: disable=pointless-statement
+    except (CommandExecutionError, AttributeError):
+        pytest.skip("Unable to install npm")
```

Comment:
```suggestion
def install_npm(sminion):
    sminion.states.pkg.installed("npm")
    if sminion.functions.npm.__virtual__() is not True:
        pytest.skip("Unable to install npm")
```

Reply:
I actually just tried `sminion.states.pkg.installed` before looking at this: it fails with this:

```
>       kwargs["saltenv"] = __env__                                                                                                                                                          
E       NameError: name '__env__' is not defined                                                                                                                                             
                                                                                                                                                                                             
salt/states/pkg.py:1692: NameError           
```

But `state.single` works.

I'm not sure if CommandExecutionError is necessary when using state.single, but on some of the platforms - specifically freebsd I think (?) may have been mac - were raising that when using pkg.install when the npm package wasn't found.

Reply:
```suggestion
def install_npm(states, modules):
    ret = states.pkg.installed("npm")
    assert ret.result is True
    if modules.npm.__virtual__() is not True:
        pytest.skip("Unable to install npm")
```

In that case, this would work as all salt dunders should be present.

Reply:
will this work if the npm module is not loaded? This is the error that I was seeing when npm wasn't installed: https://jenkins.saltproject.io/job/pr-freebsd-130-amd64-py3-pytest/job/PR-60505/4/testReport/tests.pytests.functional.states/test_npm/test_removed_installed_cycle/

Reply:
You're right, probably `"npm" in modules`, which will give you a Boolean

Reply:
Lemme give that a shot and see if it does what we want

Reply:
Looks like that doesn't:

```
        modules.npm
>       assert "npm" in modules
E       AssertionError: assert 'npm' in <salt.loader.LazyLoader object at 0x7fed67cadac0>
```

Reply:
Yeah, this is weird, the lazy loader should support that....
Nonetheless, your way works too